colossus-tests/src/test/scala/colossus/protocols/http/HttpParseSpec.scala

```diff
@@ -76,21 +76,24 @@ class HttpParserSuite extends WordSpec with MustMatchers{
     "parse request with content" in {
       val body = ByteString("HELLO I AM A BODY")
       val len = body.size
-      val req = s"POST /hello/world HTTP/1.1\r\nHost: api.foo.bar:444\r\nAccept: */*\r\nContent-Length: $len\r\n\r\n${body.utf8String}"
+      val req = s"POST /hello/world HTTP/1.1\r\nHost: api.foo.bar:444\r\nAccept: */*\r\nContent-Type: text/plain\r\nContent-Length: $len\r\n\r\n${body.utf8String}"
       val expected = HttpRequest(HttpRequestHead(
         method = HttpMethod.Post,
         url = "/hello/world",
         version = HttpVersion.`1.1`,
         headers = List(
           "host" -> "api.foo.bar:444",
           "accept" ->  "*/*",
+          "content-type" -> "text/plain",
           "content-length" -> len.toString
         )
-      ), HttpBody(body))
+      ), HttpBody(body, HttpHeader("Content-Type", "text/plain")))
```

Comment:
Hmm that's interesting. I do some several equal methods being overridden, although it's not obvious how that affects the equality of HttpRequest. For now I'd suggest creating an issue to investigate further and link to this code.